libraries/AP_Rally/AP_Rally.cpp

```diff
@@ -55,6 +55,16 @@ AP_Rally::AP_Rally(AP_AHRS &ahrs)
     AP_Param::setup_object_defaults(this, var_info);
 }
 
+// get a rally point total count from EEPROM
+uint8_t AP_Rally::get_rally_point_total_count() const {
+    return _rally_point_total_count;
+}
+
+bool AP_Rally::increment_rally_point_total_count() {
+    return AP_Param::set_by_name("RALLY_TOTAL", _rally_point_total_count + 1);
```

Comment:
Maybe I'm wrong but if I just increase `_rally_point_total_count`, it will not be saved in the EEPROM ?
I'll add a method than does increment and add a new rally point in one go, it's what I was doing in GCS_Mavlink with : 
` uint8_t new_rally_id = copter.rally.get_rally_point_total_count();
  copter.rally.increment_rally_point_total_count();
  copter.rally.set_rally_point_with_index(new_rally_id, rally_location); `

Reply:
Use `set_and_save_ifchanged()` https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Param/AP_Param.h#L674

Reply:
@Kiwa21 set_by_name doesn't save it either, which is why I suggested that was enough. As @WickedShell there are methods for setting and saving the parameter value.